core/pom.xml

```diff
@@ -596,12 +596,6 @@ THE SOFTWARE.
       <version>1.3.1-jenkins-1</version>
     </dependency>
 
-    <dependency>
```

Comment:
I was surprised that this was a deletion rather than a replacement.  I don't see any reference to connectbot elsewhere in the pom file.

I assume this means that there are no direct dependencies to jbcrypt in Jenkins core.  Otherwise, it would have failed to compile.

Removing the explicit dependency on jbcrypt 0.4 will then allow the transitive dependency from trilead to jbcrypt 1.0.0 for trilead's use.

Reply:
There _are_ usages of JBCrypt in core. They now use the transitive dependency as far as the Maven compiler plugin is concerned. From the perspective of runtime class loading, it means there is just one `jenkins.war!/WEB-INF/lib/jbcrypt-*.jar` rather than two.

Reply:
Is it a general pattern to rely on transitive dependency resolution like that?  

I assumed (simple minded) that if there was a JBCrypt usage, it would have a dependency declared.

I'm also fine with relying on the transitive dependencies, since they are required for trilead.

Reply:
I think there is no policy about it. If there were, we would apply the Enforcer rule.